combat-trainer.lic

```diff
@@ -149,13 +153,16 @@ class SetupProcess
   def check_weapon(weapon_skill)
     echo("checking weapons as #{@last_skill.inspect}!=#{weapon_skill}") if $debug_mode
 
-    if @elemental_weapons.include?(weapon_skill)
-      summon_elemental_weapon unless @elemental_weapons.include?(@last_skill)
-      shape_elemental_weapon(weapon_skill)
+    if @summoned_weapons.include?(weapon_skill)
```

Comment:
We're checking `@summoned_weapons.include?(@last_skill)` in both branches of this if. Can we commonize some of this logic?